### PR TITLE
8.0 website menu display

### PR DESCRIPTION
**Pull request by foutoucour**
The module manages display website menu entries, depending if the user is
logged or not.
The selection of the display status can be chosen logged and/or not.
Extends features and view of website.menu model.

Website.menu list view can be found at:
Settings > Configuration > Website Settings > Configure Website menus

The module inherit from website.menu to add 2 booleans fields, user_logged
and user_not_logged.
On top of that, website.layout template is extended to include a condition
that drive if the menu is built or not.
It has been choose to not only hide the menu to avoid to easily get around
by editing the html DOM.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Very interesting module! I would rename it to something with a clearer meaning: _website_menu_log_status_, because display can be for a lot of reasons to be displayed/not.

Thanks anyway for the contribution.

**Comment by pedrobaeza** [comment]
Please check a little pep8 issue.

**Comment by pedrobaeza** [comment]
Another doubt: if someone has direct URL link, he can still enter the page, doesn't it?

**Comment by max3903** [comment]
I would propose website_menu_by_user_status

**Comment by pedrobaeza** [comment]
Yes, good name, @max3903

**Comment by bjoliveau** [comment]
:+1: website_menu_by_user_status

**Comment by bjoliveau** [comment]
:+1:

File: website_menu_by_user_status/models/website_menu.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+class WebsiteMenu(models.Model):
+    """Improve website.menu with adding booleans that drive
+    if the menu is displayed when the user is logger or not.
+    """
+    _inherit = 'website.menu'
+
+    user_logged = fields.Boolean(
+        string="User Logged",
+        default=True,
+        help=_("If checked, the menu will be display when the user is logged.")
```
Comment (hbrunn) [inline website_menu_by_user_status/models/website_menu.py]: displayed

File: website_menu_by_user_status/models/website_menu.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    )
+
+    user_not_logged = fields.Boolean(
+        string="User Not Logged",
+        default=True,
+        help=_("If checked, "
+               "the menu will be display when the user is not logged.")
```
Comment (hbrunn) [inline website_menu_by_user_status/models/website_menu.py]: displayed

File: website_menu_by_user_status/views/website_templates.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <template id="website_display_menu" name="Conditional display menu"
+                  inherit_id="website.layout"
+                  active="True" customize_show="True">
+            <xpath expr="//ul[@id='top_menu']/t/t" position="replace">
+                <t t-if="(submenu.user_logged and request.uid != 3) or (submenu.user_not_logged and request.uid == 3)">
```
Comment (hbrunn) [inline website_menu_by_user_status/views/website_templates.xml]: Don't use a number here (that can change per database) but `%(base.public_user)d`

**Comment by foutoucour** [comment]
Could `needs fixing` label removed? I think everything is green here now :)

**Pull request merged**